web/app/scripts/leaflet/leaflet-defaults-provider.js

```diff
@@ -0,0 +1,47 @@
+(function() {
+    'use strict';
+
+    /**
+     * Provides defaults for new maps created using the leafletMap directive.
+     *
+     * Defaults can be changed at config time via LeafletDefaults.setDefaults()
+     *
+    /* ngInject */
+    function LeafletDefaultsProvider() {
+        var defaults = {
+            center: [0,0],
+            zoom: 1,
+            crs: L.CRS.EPSG4326
+        };
+
+        /**
+         * Update defaults by merging user-set defaults into defaults object
+         * @param {LeafletDefaults} newDefaults
+         */
+        this.setDefaults = function (newDefaults) {
```

Comment:
I like the pattern we use in all the other modules, where `this` is set to a variable at the top of the function, and then the variable is used instead of `this`. It gets us away from needing to worry about the context-aware intricacies of `this`.